File: VB/WindowsApplication3/MyDockManager.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraBars.Docking
Imports System.Windows.Forms
Imports System.ComponentModel

Namespace WindowsApplication3
	Public Class MyDockManager
		Inherits DockManager
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal form As ContainerControl)
			MyBase.New(form)
		End Sub
		Public Sub New(ByVal container As IContainer)
			MyBase.New(container)
		End Sub

		Protected Overrides Function CreateVisualizer(ByVal role As VisualizerRole) As VS2005StyleDockingVisualizer
			If (Not SupportVS2010DockingStyle()) Then
				Return New MyVS2005StyleDockingVisualizer(role, DockModeVS2005FadeSpeed, DockModeVS2005FadeFramesCount, Me)
			End If
			Return MyBase.CreateVisualizer(role)
		End Function
	End Class

	Public Class MyVS2005StyleDockingVisualizer
		Inherits VS2005StyleDockingVisualizer
		Private manager As DockManager
		Public Sub New(ByVal role As VisualizerRole, ByVal fadeSpeed As Integer, ByVal framesCount As Integer, ByVal manager As DockManager)
			MyBase.New(role, fadeSpeed, framesCount)
			Me.manager = manager
		End Sub

		Protected Overrides Sub UpdateWindows()
			Dim panel As DockPanel = manager.ActivePanel
			If panel.Options.AllowDockBottom Then
				SetBoundsVisibility(GetFrame(VisualizerFormType.Bottom), Me.viewInfoCore(VisualizerFormType.Bottom))
			End If
			If panel.Options.AllowDockFill Then
				SetBoundsVisibility(GetFrame(VisualizerFormType.Center), Me.viewInfoCore(VisualizerFormType.Center))
			End If
			If panel.Options.AllowDockLeft Then
				SetBoundsVisibility(GetFrame(VisualizerFormType.Left), Me.viewInfoCore(VisualizerFormType.Left))
			End If
			If panel.Options.AllowDockRight Then
				SetBoundsVisibility(GetFrame(VisualizerFormType.Right), Me.viewInfoCore(VisualizerFormType.Right))
			End If
			If panel.Options.AllowDockTop Then
				SetBoundsVisibility(GetFrame(VisualizerFormType.Top), Me.viewInfoCore(VisualizerFormType.Top))
			End If
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication3/MyDockManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraBars.Docking;
using System.Windows.Forms;
using System.ComponentModel;

namespace WindowsApplication3
{
    public class MyDockManager : DockManager
    {
        public MyDockManager() : base() { }
        public MyDockManager(ContainerControl form) : base(form) { }
        public MyDockManager(IContainer container) : base(container) { }

        protected override VS2005StyleDockingVisualizer CreateVisualizer(VisualizerRole role)
        {
            if (!SupportVS2010DockingStyle())
                return new MyVS2005StyleDockingVisualizer(role, DockModeVS2005FadeSpeed, DockModeVS2005FadeFramesCount, this);
            return base.CreateVisualizer(role);
        }
    }

    public class MyVS2005StyleDockingVisualizer : VS2005StyleDockingVisualizer
    {
        DockManager manager;
        public MyVS2005StyleDockingVisualizer(VisualizerRole role, int fadeSpeed, int framesCount, DockManager manager)
            : base(role, fadeSpeed, framesCount)
        {
            this.manager = manager;
        }

        protected override void UpdateWindows()
        {
            DockPanel panel = manager.ActivePanel;
            if (panel.Options.AllowDockBottom)
                SetBoundsVisibility(GetFrame(VisualizerFormType.Bottom), this.viewInfoCore[VisualizerFormType.Bottom]);
            if (panel.Options.AllowDockFill)
                SetBoundsVisibility(GetFrame(VisualizerFormType.Center), this.viewInfoCore[VisualizerFormType.Center]);
            if (panel.Options.AllowDockLeft)
                SetBoundsVisibility(GetFrame(VisualizerFormType.Left), this.viewInfoCore[VisualizerFormType.Left]);
            if (panel.Options.AllowDockRight)
                SetBoundsVisibility(GetFrame(VisualizerFormType.Right), this.viewInfoCore[VisualizerFormType.Right]);
            if (panel.Options.AllowDockTop)
                SetBoundsVisibility(GetFrame(VisualizerFormType.Top), this.viewInfoCore[VisualizerFormType.Top]);
        }
    }
}
